hcloud/client.go

```
@@ -169,15 +179,24 @@ func (c *Client) Do(r *http.Request, v interface{}) (*Response, error) {
 			return nil, err
 		}
 		response := &Response{Response: resp}
-
 		body, err := ioutil.ReadAll(resp.Body)
 		if err != nil {
 			resp.Body.Close()
 			return response, err
 		}
 		resp.Body.Close()
 		resp.Body = ioutil.NopCloser(bytes.NewReader(body))
-
+		if c.debugWriter != nil {
+			fmt.Fprintln(c.debugWriter, "------- HCLOUD_GO_DEBUG -------")
+			// Print Request Details
+			dumpReq, _ := httputil.DumpRequest(r, true)
```

I don’t think you should ignore errors here. Docs:

>  If DumpRequest returns an error, the state of req is undefined.